Model/Api/UpdateDiscountTrait.php

```diff
@@ -326,6 +336,16 @@ protected function applyingCouponCode($couponCode, $coupon, $quote, $addQuote =
 
         $this->discountHelper->setCouponCode($quote, $couponCode);
 
+        if ($quote->getCouponCode() === '') {
+            $this->logHelper->addInfoLog('Error: quote does not meet rule\'s conditions.');
+            throw new BoltException(
+                __('The quote does not meet rule\'s conditions of coupon code %1.', $couponCode),
+                null,
+                BoltErrorResponse::ERR_CODE_INVALID,
+                $quote
+            );
+        }
+        
         if ($quote->getCouponCode() != $couponCode) {
```

Comment:
Can we add `$quote->getCouponCode()` into bugsnag? Or we already have it?

Reply:
Done